#takes input from streamlined_whole_seqs_script_eta.py (updated 6/25/20)

library(dplyr)
library(mclust)
library(mixtools)

CpG_stat = function(data,species,output){
  data1 = read.table(data, header = TRUE, sep = " ", quote = "")
  data1 = filter(data1, Length >= 300 & CGo_e <= 2)
  data1 = data1[data1$TGo_e!="Inf" & data1$eta!="Inf",]
  g1 = na.omit(data1$CGo_e)
  mean = mean(g1)
  g1_clust = Mclust(g1)
  LinearModel = lm(data1$TGo_e ~ data1$CGo_e)
  pdf(file = paste(species, 'plot1'), width = 120, height = 50)
  par(mfrow=c(2,3))
  par(mai = c(6,8,6,1))
  par(cex = 7)
  plot(density(g1), main = species, xlab = "Normalized CpG content", ylab = "Density", lwd = 10)
  plot(x = data1$CGo_e, y = data1$TGo_e, xlim = c(0,2), ylim = c(0,3), xlab = "Normalized CpG content", ylab = "Normalized TpG content", pch = 20)
  abline(LinearModel, col = "red", lwd = 8)
  plot(density(na.omit(data1$eta)), xlab = "eta", ylab = "Density", xlim = c(0,5),lwd = 10, main = "")
  plot(x = data1$CGo_e, y = data1$eta, xlim = c(0,2), pch = 20, xlab = "Normalized CpG content", ylab ="eta")
  plot(g1_clust, what = "BIC", lwd = 10)
  cat(paste(species, "\t", mean, "\t", coef(LinearModel)[1], "\t", coef(LinearModel)[2], "\t", g1_clust$modelName, g1_clust$G, "\n"), file=output, append = TRUE)
  dev.off()
}

### some random testing code ### 
  data1 = read.table("Aurelia.Genome_repeats_nonoverlapping_with_gene_models_eta_whole_seqs2.txt", header = TRUE, sep = " ", quote = "")
  data1 = filter(data1, Length >= 300 & CGo_e <= 2)
  data1 = data1[data1$TGo_e!="Inf" & data1$eta!="Inf" & data1$CGo_e != "Inf",]
  which(data1$CGo_e == "Inf")
  g1 = na.omit(data1$CGo_e)
  mean = mean(g1)
  g1_clust = Mclust(g1)
  LinearModel = lm(data1$TGo_e ~ data1$CGo_e)
  pdf(file = paste("Aurelia coerulia repeats nonoverlapping with genes", 'plot1'), width = 120, height = 50)
  par(mfrow=c(2,3))
  par(mai = c(6,8,6,1))
  par(cex = 7)
  plot(density(g1), main = "Aurelia coerulia (repeats)", xlab = "Normalized CpG content", ylab = "Density", lwd = 10, xlim = c(0,2))
  plot(x = data1$CGo_e, y = data1$TGo_e, xlim = c(0,2), ylim = c(0,3), xlab = "Normalized CpG content", ylab = "Normalized TpG content", pch = 20)
  abline(LinearModel, col = "red", lwd = 8)
  plot(density(na.omit(data1$eta)), xlab = "eta", ylab = "Density", xlim = c(0,5),lwd = 10, main = "")
  plot(x = data1$CGo_e, y = data1$eta, xlim = c(0,2), pch = 20, xlab = "Normalized CpG content", ylab ="eta")
  plot(g1_clust, what = "BIC", lwd = 10)
  cat(paste("Aurelia coerulia (repeats subtract gene models)", "\t", mean, "\t", coef(LinearModel)[1], "\t", coef(LinearModel)[2], "\t", g1_clust$modelName, g1_clust$G, "\n"), file="Cnidarian_stats.txt", append = TRUE)
  dev.off()
###




setwd("/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta")
cat(paste("species", "\t", "Mean_GB_CpG", "\t", "lm(TpG~CpG)_Intersect","\t", "lm(TpG~CpG)_Slope", "\t", "Gaussian_model_name", "#Gaussian_models", "\n"), file='Cnidarian_stats.txt', append = TRUE)

CpG_stat(data="Abylopsis_tetragona_eta_whole_seqs2.txt", species = "Abylopsis tetragona", output = "Cnidarian_stats.txt")
CpG_stat(data="Acanthogorgia_aspera_GETB01_eta_whole_seqs2.txt", species = "Acanthogorgia aspera", output = "Cnidarian_stats.txt")
CpG_stat(data="Aegina_citrea_eta_whole_seqs2.txt", species = "Aegina citrea", output = "Cnidarian_stats.txt")
CpG_stat(data="Agalma_elegans_eta_whole_seqs2.txt", species = "Agalma elegans", output = "Cnidarian_stats.txt")
CpG_stat(data="Aiptasia_pallida_eta_whole_seqs2.txt", species = "Aiptasia pallida", output = "Cnidarian_stats.txt")
CpG_stat(data="Alatina_alata_eta_whole_seqs2.txt", species = "Alatina alata", output = "Cnidarian_stats.txt")
CpG_stat(data="Anthopleura_elegantissima_SRR1645256_eta_whole_seqs2.txt", species = "Anthopleura elegantissima", output = "Cnidarian_stats.txt")
CpG_stat(data="Atolla_vanhoeffeni_eta_whole_seqs2.txt", species = "Atolla vanhoeffeni", output = "Cnidarian_stats.txt")
CpG_stat(data="Aurelia_aurita_GBRG00000000_eta_whole_seqs2.txt", species = "Aurelia aurita", output = "Cnidarian_stats.txt")
CpG_stat(data="Aurelia_eta_whole_seqs2.txt", species = "Aurelia coerulia (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data="Aurelia_coerulia_repeats_eta_whole_seqs2.txt", species = "Aurelia coerulia (repeats)", output = "Cnidarian_stats.txt")
CpG_stat(data="Briareum_asbestinum_GHBD02_eta_whole_seqs2.txt", species = "Briareum asbestinum", output = "Cnidarian_stats.txt")
CpG_stat(data="Calvadosia_cruxmelitensis_eta_whole_seqs2.txt", species = "Calvadosia cruxmelitensis", output = "Cnidarian_stats.txt")
CpG_stat(data="Chironex_fleckeri_eta_whole_seqs2.txt", species = "Chironex fleckeri", output = "Cnidarian_stats.txt")
CpG_stat(data="Chironex_yamaguchii_GHAX01_eta_whole_seqs2.txt", species = "Chironex yamaguchii", output = "Cnidarian_stats.txt")
CpG_stat(data="Chrysaora_fuscescens_SRR3180892_eta_whole_seqs2.txt", species = "Chrysaora fuscescens", output = "Cnidarian_stats.txt")
CpG_stat(data="Clavularia_sp_GHAW01_eta_whole_seqs2.txt", species = "Clavularia sp", output = "Cnidarian_stats.txt")
CpG_stat(data="Clytia_hemisphaerica_HAMU01_eta_whole_seqs2.txt", species = "Clytia hemisphaerica", output = "Cnidarian_stats.txt")
CpG_stat(data="Copula_sivickisi_GHBG01_eta_whole_seqs2.txt", species = "Copula sivickisi", output = "Cnidarian_stats.txt")
CpG_stat(data="Corallium_rubrum_SRR1552944_eta_whole_seqs2.txt", species = "Corallium rubrum", output = "Cnidarian_stats.txt")
CpG_stat(data="Corynactis_australis_SRR3198576-79-80_eta_whole_seqs2.txt", species = "Corynactis australis", output = "Cnidarian_stats.txt")
CpG_stat(data="Craseoa_lathetica_eta_whole_seqs2.txt", species = "Craseoa lathetica", output = "Cnidarian_stats.txt")
CpG_stat(data="Craspedacusta_sowerbyi_SRR923472_eta_whole_seqs2.txt", species = "Craspedacusta sowerbyi", output = "Cnidarian_stats.txt")
CpG_stat(data="Craterolophus_convolvulus_eta_whole_seqs2.txt", species = "Craterolophus convolvulus", output = "Cnidarian_stats.txt")
CpG_stat(data="Ctenactis_echinata_SRR2566726_eta_whole_seqs2.txt", species = "Ctenactis echinata", output = "Cnidarian_stats.txt")
CpG_stat(data="Dynamena_pumila_GHMC01_eta_whole_seqs2.txt", species = "Dynamena pumila", output = "Cnidarian_stats.txt")
CpG_stat(data="Ectopleura_larynx_eta_whole_seqs2.txt", species = "Ectopleura larynx", output = "Cnidarian_stats.txt")
CpG_stat(data="Edwardsiella_lineata_ElT1_eta_whole_seqs2.txt", species = "Edwardsiella lineata", output = "Cnidarian_stats.txt")
CpG_stat(data="Eleutherobia_rubra_GHFI01_eta_whole_seqs2.txt", species = "Eleutherobia rubra", output = "Cnidarian_stats.txt")
CpG_stat(data="Eunicella_cavolinii_SRR1324943_eta_whole_seqs2.txt", species = "Eunicella cavolinii", output = "Cnidarian_stats.txt")
CpG_stat(data="Eunicella_verrucosa_SRR1324944_eta_whole_seqs2.txt", species = "Eunicella verrucosa", output = "Cnidarian_stats.txt")
CpG_stat(data="Favia_lizardensis_SRR2566729_eta_whole_seqs2.txt", species = "Favia lizardensis", output = "Cnidarian_stats.txt")
CpG_stat(data="Gorgonia_ventalina_SRR935083_eta_whole_seqs2.txt", species = "Gorgonia ventalina", output = "Cnidarian_stats.txt")
CpG_stat(data="Haliclystus_auricula_HAHA01_eta_whole_seqs2.txt", species = "Haliclystus auricula", output = "Cnidarian_stats.txt")
CpG_stat(data="Haliclystus_sanjuanensis_eta_whole_seqs2.txt", species = "Haliclystus sanjuanensis", output = "Cnidarian_stats.txt")
CpG_stat(data="Heliopora_coerulea_GFVH01_eta_whole_seqs2.txt", species = "Heliopora coerulea", output = "Cnidarian_stats.txt")
CpG_stat(data="Hydra_oligactis_SRX019577_eta_whole_seqs2.txt", species = "Hydra oligactis", output = "Cnidarian_stats.txt")
CpG_stat(data="Hydra_viridissima_SRX019578_eta_whole_seqs2.txt", species = "Hydra viridissima", output = "Cnidarian_stats.txt")
CpG_stat(data="Hydractinia_polyclina_SRR923509_eta_whole_seqs2.txt", species = "Hydractinia polyclina", output = "Cnidarian_stats.txt")
CpG_stat(data="Hydractinia_symbiolongicarpus_eta_whole_seqs2.txt", species = "Hydractinia symbiolongicarpus", output = "Cnidarian_stats.txt")
CpG_stat(data="Kudoa_iwatai_eta_whole_seqs2.txt", species = "Kudoa iwatai", output = "Cnidarian_stats.txt")
CpG_stat(data="Leptogorgia_sarmentosa_SRR1324968_eta_whole_seqs2.txt", species = "Leptogorgia sarmentosa", output = "Cnidarian_stats.txt")
CpG_stat(data="Leucernaria_quadricornis_eta_whole_seqs2.txt", species = "Leucernaria quadricornis", output = "Cnidarian_stats.txt")
CpG_stat(data="Lobactis_scutaria_SRR2300562_eta_whole_seqs2.txt", species = "Lobactis scutaria", output = "Cnidarian_stats.txt")
CpG_stat(data="Madracis_auretenra_Maur_eta_whole_seqs2.txt", species = "Madracis auretenra", output = "Cnidarian_stats.txt")
CpG_stat(data="Montastraea_cavernosa_SRR2306543_eta_whole_seqs2.txt", species = "Montastraea cavernosa", output = "Cnidarian_stats.txt")
CpG_stat(data="Myxobolus_cerebralis_eta_whole_seqs2.txt", species = "Myxobolus cerebralis", output = "Cnidarian_stats.txt")
CpG_stat(data="Myxobolus_pendula_SRX1269149_eta_whole_seqs2.txt", species = "Myxobolus pendula", output = "Cnidarian_stats.txt")
CpG_stat(data="Nanomia_bijuga_eta_whole_seqs2.txt", species = "Nanomia bijuga", output = "Cnidarian_stats.txt")
CpG_stat(data="Physalia_physalis_eta_whole_seqs2.txt", species = "Physalia physalis", output = "Cnidarian_stats.txt")
CpG_stat(data="Platygyra_carnosus_SRX118069_eta_whole_seqs2.txt", species = "Platygyra carnosus", output = "Cnidarian_stats.txt")
CpG_stat(data="Podocoryna_carnea_eta_whole_seqs2.txt", species = "Podocoryna carnea", output = "Cnidarian_stats.txt")
CpG_stat(data="Polypodum_hydriforme_SRR1336770_eta_whole_seqs2.txt", species = "Polypodum hydriforme", output = "Cnidarian_stats.txt")
CpG_stat(data="Porpita_porpita_GHBA01_eta_whole_seqs2.txt", species = "Porpita_porpita", output = "Cnidarian_stats.txt")
CpG_stat(data="Renilla_reniformis_eta_whole_seqs2.txt", species = "Renilla reniformis", output = "Cnidarian_stats.txt")
CpG_stat(data="Rhodactis_indosinensis_SRR3201278-81_eta_whole_seqs2.txt", species = "Rhodactis indosinensis", output = "Cnidarian_stats.txt")
CpG_stat(data="Rhopilema_esculentum_GEMS01_eta_whole_seqs2.txt", species = "Rhopilema esculentum", output = "Cnidarian_stats.txt")
CpG_stat(data="Ricordea_yuma_SRR3201251-3_eta_whole_seqs2.txt", species = "Ricordea yuma", output = "Cnidarian_stats.txt")
CpG_stat(data="Seriatopora_hystrix_SRR2300678_eta_whole_seqs2.txt", species = "Seriatopora hystrix", output = "Cnidarian_stats.txt")
CpG_stat(data="Stomolophus_meleagris_eta_whole_seqs2.txt", species = "Stomolophus meleagris", output = "Cnidarian_stats.txt")
CpG_stat(data="Tripedalia_cystophora_GHAQ01_eta_whole_seqs2.txt", species = "Tripedalia cystophora", output = "Cnidarian_stats.txt")
CpG_stat(data="Turritopsis_sp_SK-2016_DRR053671-DRR053676_eta_whole_seqs2.txt", species = "Turritopsis sp.", output = "Cnidarian_stats.txt")
CpG_stat(data="Velella_velella_GHAZ01_eta_whole_seqs2.txt", species = "Velella velella", output = "Cnidarian_stats.txt")
CpG_stat(data="Xenia_sp_GHBC01_eta_whole_seqs2.txt", species = "Xenia sp.", output = "Cnidarian_stats.txt")
CpG_stat(data="Nemopilema_nomurai_GHAR01_eta_whole_seqs2.txt", species = "Nemopilema nomurai", output = "Cnidarian_stats.txt")
CpG_stat(data="Cxam_eta_whole_seqs2.txt", species = "Cassiopea xamachana", output = "Cnidarian_stats.txt")
CpG_stat(data="Nematostella_vectensis_GCF_000209225.1_ASM20922v1_rna_eta_whole_seqs2.txt", species = "Nematostella vectensis", output = "Cnidarian_stats.txt")
CpG_stat(data = "Cassiopea_xamachana_transcriptome_eta_whole_seqs2.txt", species = "Cassiopea xamachana (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Acropora_digitefera_GCF_000222465_rna_eta_whole_seqs.txt", species = "Acropora digitefera (transcriptome)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
CpG_stat(data = "Acropora_millepora_GCF_004143615_rna_eta_whole_seqs.txt", species = "Acropora millepora (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Actinia_tenebrosa_GCF_009602425_eta_whole_seqs.txt", species = "Actinia tenebrosa (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Actinia_tenebrosa_GCF_009602425_rna_eta_whole_seqs.txt", species = "Actinia tenebrosa (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Aiptasia_pallida_GCF_001417965_eta_whole_seqs.txt", species = "Aiptasia pallida (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Aiptasia_pallida_GCF_001417965_rna_eta_whole_seqs.txt", species = "Aiptasia pallida (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Dendronephthya_gigantea_GCF_004324835_eta_whole_seqs.txt", species = "Dendronephthya gigantea (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Dendronephthya_gigantea_GCF_004324835_rna_eta_whole_seqs.txt", species = "Dendronephthya gigantea (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Henneguya_salminicola_GCA_009887335_eta_whole_seqs.txt", species = "Henneguya salminicola (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Hydra_vulgaris_GCF_000004095_rna_eta_whole_seqs.txt", species = "Hydra vulgaris (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Myxobolus_squamalis_GCA_010108815_eta_whole_seqs.txt", species = "Myxobolus squamalis (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Orbicella_faveolata_GCF_002042975_eta_whole_seqs.txt", species = "Orbicella faveolata (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Orbicella_faveolata_GCF_002042975_rna_eta_whole_seqs.txt", species = "Orbicella faveolata (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Pocillopora_damicornis_GCF_003704095_eta_whole_seqs.txt", species = "Pocillopora damicornis (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Pocillopora_damicornis_GCF_003704095_rna_eta_whole_seqs.txt", species = "Pocillopora damicornis (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Stylophora_pistillata_GCF_002571385_eta_whole_seqs.txt", species = "Stylophora pistillata (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Stylophora_pistillata_GCF_002571385_rna_eta_whole_seqs.txt", species = "Stylophora pistillata (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Thelohanellus_kitauei_GCA_000827895_eta_whole_seqs.txt", species = "Thelohanellus kitauei (gene models)", output = "Cnidarian_stats.txt")
CpG_stat(data = "Morbakka_virulenta_GHAF01.1_rna_eta_whole_seqs.txt", species = "Morbakka virulenta", output = "Cnidarian_stats.txt")
CpG_stat(data = "Myxobolus_squamalis_GHBR01.1.fsa_nt_eta_whole_seqs.txt", species = "Myxobolus squamalis (transcriptome)", output = "Cnidarian_stats.txt")
CpG_stat(data = "MOR05_r06_mRNA_whole_seqs_eta.txt", species = "Morbakka virulenta (mRNA)", output = "Cnidarian_stats.txt")




#for repeats#
CpG_stat1 = function(data,species,output){
  data1 = read.table(data, header = TRUE, sep = " ", quote = "",comment.char = "")
  data1 = filter(data1, Length >= 50)
  data1 = data1[data1$TGo_e!="Inf" & data1$eta!="Inf",]
  g1 = na.omit(data1$CGo_e)
  mean = mean(g1)
  g1_clust = Mclust(g1)
  LinearModel = lm(data1$TGo_e ~ data1$CGo_e)
  pdf(file = paste(species, 'plot1'), width = 120, height = 50)
  par(mfrow=c(2,3))
  par(mai = c(6,8,6,1))
  par(cex = 7)
  plot(density(g1), main = species, xlab = "Normalized CpG content", ylab = "Density", lwd = 10)
  plot(x = data1$CGo_e, y = data1$TGo_e, xlim = c(0,2), ylim = c(0,3), xlab = "Normalized CpG content", ylab = "Normalized TpG content", pch = 20)
  abline(LinearModel, col = "red", lwd = 8)
  plot(density(na.omit(data1$eta)), xlab = "eta", ylab = "Density", xlim = c(0,5),lwd = 10, main = "")
  plot(x = data1$CGo_e, y = data1$eta, xlim = c(0,2), pch = 20, xlab = "Normalized CpG content", ylab ="eta")
  plot(g1_clust, what = "BIC", lwd = 10)
  cat(paste(species, "\t", mean, "\t", coef(LinearModel)[1], "\t", coef(LinearModel)[2], "\t", g1_clust$modelName, g1_clust$G, "\n"), file=output, append = TRUE)
  dev.off()
}
CpG_stat1(data = "Heliopora_coerulea_GFVH01_eta_whole_seqs2.txt", species = "Morbakka virulenta (repeats)", output = "Cnidarian_stats.txt")


### Model fitting ## 

#random testing codes#
data1 = read.table("Aiptasia_pallida_eta_whole_seqs2.txt", header = TRUE, sep = " ", quote = "")
CG = data1$CGo_e[data1$CGo_e >= 0.0001 & data1$CGo_e <= 1.5]
data1_model = normalmixEM(CG)
plot(data1_model, which = 2, col2 = c("red", "blue"), main2 = "Aiptasia_pallida", xlab2 = "", mai = c(0.5,1,0.5,0.5))


mixmodel = function(data,species,output){
  data1 = read.table(data, header = TRUE, sep = " ", quote = "")
  data1 = filter(data1, Length >= 300 & CGo_e <= 1.5 & CGo_e >= 0.0001)
  CGmodel = normalmixEM(data1$CGo_e)
  plot(CGmodel, which = 2, col2 = c("red", "blue"), main2= species, xlab2 = "")
  cat(paste(species, "\t", CGmodel$lambda[1], "\t", CGmodel$lambda[2], "\t", CGmodel$mu[1], "\t", CGmodel$mu[2], CGmodel$sigma[1], "\t", CGmodel$sigma[2], CGmodel$loglik, "\n"), file=output, append = TRUE)
}
cat(paste("\n", "species", "\t", "lambda1", "\t", "lambda2","\t", "mu1", "\t", "mu2", "\t", "sigma1", "\t", "sigma2","\t","loglik", "\n"), file='Cnidarian_stats.txt', append = TRUE)
par(mfrow=c(4,4), mar = c(1,2,4,0), oma = c(2,2,2,0), mgp = c(2,1,0))

mixmodel(data="Abylopsis_tetragona_eta_whole_seqs2.txt", species = "Abylopsis tetragona", output = "Cnidarian_stats.txt")
mixmodel(data="Acanthogorgia_aspera_GETB01_eta_whole_seqs2.txt", species = "Acanthogorgia aspera", output = "Cnidarian_stats.txt")
mixmodel(data = "Acropora_digitefera_GCF_000222465_rna_eta_whole_seqs.txt", species = "Acropora digitefera", output = "Cnidarian_stats.txt")
mixmodel(data = "Acropora_millepora_GCF_004143615_rna_eta_whole_seqs.txt", species = "Acropora millepora", output = "Cnidarian_stats.txt")
mixmodel(data = "Actinia_tenebrosa_GCF_009602425_rna_eta_whole_seqs.txt", species = "Actinia tenebrosa", output = "Cnidarian_stats.txt")
mixmodel(data="Aegina_citrea_eta_whole_seqs2.txt", species = "Aegina citrea", output = "Cnidarian_stats.txt")
mixmodel(data="Agalma_elegans_eta_whole_seqs2.txt", species = "Agalma elegans", output = "Cnidarian_stats.txt")
mixmodel(data="Aiptasia_pallida_eta_whole_seqs2.txt", species = "Aiptasia pallida", output = "Cnidarian_stats.txt")
mixmodel(data="Alatina_alata_eta_whole_seqs2.txt", species = "Alatina alata", output = "Cnidarian_stats.txt")
mixmodel(data="Anthopleura_elegantissima_SRR1645256_eta_whole_seqs2.txt", species = "Anthopleura elegantissima", output = "Cnidarian_stats.txt")
mixmodel(data="Atolla_vanhoeffeni_eta_whole_seqs2.txt", species = "Atolla vanhoeffeni", output = "Cnidarian_stats.txt")
mixmodel(data="Aurelia_aurita_GBRG00000000_eta_whole_seqs2.txt", species = "Aurelia aurita", output = "Cnidarian_stats.txt")
mixmodel(data="Aurelia_eta_whole_seqs2.txt", species = "Aurelia coerulia", output = "Cnidarian_stats.txt")
mixmodel(data="Briareum_asbestinum_GHBD02_eta_whole_seqs2.txt", species = "Briareum asbestinum", output = "Cnidarian_stats.txt")
mixmodel(data="Calvadosia_cruxmelitensis_eta_whole_seqs2.txt", species = "Calvadosia cruxmelitensis", output = "Cnidarian_stats.txt")
mixmodel(data = "Cassiopea_xamachana_transcriptome_eta_whole_seqs2.txt", species = "Cassiopea xamachana", output = "Cnidarian_stats.txt")

mixmodel(data="Chironex_fleckeri_eta_whole_seqs2.txt", species = "Chironex fleckeri", output = "Cnidarian_stats.txt")
mixmodel(data="Chironex_yamaguchii_GHAX01_eta_whole_seqs2.txt", species = "Chironex yamaguchii", output = "Cnidarian_stats.txt")
mixmodel(data="Chrysaora_fuscescens_SRR3180892_eta_whole_seqs2.txt", species = "Chrysaora fuscescens", output = "Cnidarian_stats.txt")
mixmodel(data="Clavularia_sp_GHAW01_eta_whole_seqs2.txt", species = "Clavularia sp", output = "Cnidarian_stats.txt")
mixmodel(data="Clytia_hemisphaerica_HAMU01_eta_whole_seqs2.txt", species = "Clytia hemisphaerica", output = "Cnidarian_stats.txt")
mixmodel(data="Copula_sivickisi_GHBG01_eta_whole_seqs2.txt", species = "Copula sivickisi", output = "Cnidarian_stats.txt")
mixmodel(data="Corallium_rubrum_SRR1552944_eta_whole_seqs2.txt", species = "Corallium rubrum", output = "Cnidarian_stats.txt")
mixmodel(data="Corynactis_australis_SRR3198576-79-80_eta_whole_seqs2.txt", species = "Corynactis australis", output = "Cnidarian_stats.txt")
mixmodel(data="Craseoa_lathetica_eta_whole_seqs2.txt", species = "Craseoa lathetica", output = "Cnidarian_stats.txt")
mixmodel(data="Craspedacusta_sowerbyi_SRR923472_eta_whole_seqs2.txt", species = "Craspedacusta sowerbyi", output = "Cnidarian_stats.txt")
mixmodel(data="Craterolophus_convolvulus_eta_whole_seqs2.txt", species = "Craterolophus convolvulus", output = "Cnidarian_stats.txt")
mixmodel(data="Ctenactis_echinata_SRR2566726_eta_whole_seqs2.txt", species = "Ctenactis echinata", output = "Cnidarian_stats.txt")
mixmodel(data = "Dendronephthya_gigantea_GCF_004324835_rna_eta_whole_seqs.txt", species = "Dendronephthya gigantea (transcriptome)", output = "Cnidarian_stats.txt")
mixmodel(data="Dynamena_pumila_GHMC01_eta_whole_seqs2.txt", species = "Dynamena pumila", output = "Cnidarian_stats.txt")
mixmodel(data="Ectopleura_larynx_eta_whole_seqs2.txt", species = "Ectopleura larynx", output = "Cnidarian_stats.txt")
mixmodel(data="Edwardsiella_lineata_ElT1_eta_whole_seqs2.txt", species = "Edwardsiella lineata", output = "Cnidarian_stats.txt")

mixmodel(data="Eleutherobia_rubra_GHFI01_eta_whole_seqs2.txt", species = "Eleutherobia rubra", output = "Cnidarian_stats.txt")
mixmodel(data="Eunicella_cavolinii_SRR1324943_eta_whole_seqs2.txt", species = "Eunicella cavolinii", output = "Cnidarian_stats.txt")
mixmodel(data="Eunicella_verrucosa_SRR1324944_eta_whole_seqs2.txt", species = "Eunicella verrucosa", output = "Cnidarian_stats.txt")
mixmodel(data="Favia_lizardensis_SRR2566729_eta_whole_seqs2.txt", species = "Favia lizardensis", output = "Cnidarian_stats.txt")
mixmodel(data="Gorgonia_ventalina_SRR935083_eta_whole_seqs2.txt", species = "Gorgonia ventalina", output = "Cnidarian_stats.txt")
mixmodel(data="Haliclystus_auricula_HAHA01_eta_whole_seqs2.txt", species = "Haliclystus auricula", output = "Cnidarian_stats.txt")
mixmodel(data="Haliclystus_sanjuanensis_eta_whole_seqs2.txt", species = "Haliclystus sanjuanensis", output = "Cnidarian_stats.txt")
mixmodel(data="Heliopora_coerulea_GFVH01_eta_whole_seqs2.txt", species = "Heliopora coerulea", output = "Cnidarian_stats.txt")
mixmodel(data = "Henneguya_salminicola_GCA_009887335_eta_whole_seqs.txt", species = "Henneguya salminicola (gene models)", output = "Cnidarian_stats.txt")
mixmodel(data="Hydra_oligactis_SRX019577_eta_whole_seqs2.txt", species = "Hydra oligactis", output = "Cnidarian_stats.txt")
mixmodel(data="Hydra_viridissima_SRX019578_eta_whole_seqs2.txt", species = "Hydra viridissima", output = "Cnidarian_stats.txt")
mixmodel(data = "Hydra_vulgaris_GCF_000004095_rna_eta_whole_seqs.txt", species = "Hydra vulgaris (transcriptome)", output = "Cnidarian_stats.txt")
mixmodel(data="Hydractinia_polyclina_SRR923509_eta_whole_seqs2.txt", species = "Hydractinia polyclina", output = "Cnidarian_stats.txt")
mixmodel(data="Hydractinia_symbiolongicarpus_eta_whole_seqs2.txt", species = "Hydractinia symbiolongicarpus", output = "Cnidarian_stats.txt")
mixmodel(data="Kudoa_iwatai_eta_whole_seqs2.txt", species = "Kudoa iwatai", output = "Cnidarian_stats.txt")
mixmodel(data="Leptogorgia_sarmentosa_SRR1324968_eta_whole_seqs2.txt", species = "Leptogorgia sarmentosa", output = "Cnidarian_stats.txt")


mixmodel(data="Leucernaria_quadricornis_eta_whole_seqs2.txt", species = "Leucernaria quadricornis", output = "Cnidarian_stats.txt")
mixmodel(data="Lobactis_scutaria_SRR2300562_eta_whole_seqs2.txt", species = "Lobactis scutaria", output = "Cnidarian_stats.txt")
mixmodel(data="Madracis_auretenra_Maur_eta_whole_seqs2.txt", species = "Madracis auretenra", output = "Cnidarian_stats.txt")
mixmodel(data="Montastraea_cavernosa_SRR2306543_eta_whole_seqs2.txt", species = "Montastraea cavernosa", output = "Cnidarian_stats.txt")
mixmodel(data="MOR05_r06_mRNA_whole_seqs_eta.txt", species = "Morbakka virulenta", output = "Cnidarian_stats.txt")
mixmodel(data="Myxobolus_cerebralis_eta_whole_seqs2.txt", species = "Myxobolus cerebralis", output = "Cnidarian_stats.txt")
mixmodel(data="Myxobolus_pendula_SRX1269149_eta_whole_seqs2.txt", species = "Myxobolus pendula", output = "Cnidarian_stats.txt")
mixmodel(data = "Myxobolus_squamalis_GHBR01.1.fsa_nt_eta_whole_seqs.txt", species = "Myxobolus squamalis", output = "Cnidarian_stats.txt")
mixmodel(data="Nanomia_bijuga_eta_whole_seqs2.txt", species = "Nanomia bijuga", output = "Cnidarian_stats.txt")
mixmodel(data="Nematostella_vectensis_GCF_000209225.1_ASM20922v1_rna_eta_whole_seqs2.txt", species = "Nematostella vectensis", output = "Cnidarian_stats.txt")
mixmodel(data="Nemopilema_nomurai_GHAR01_eta_whole_seqs2.txt", species = "Nemopilema nomurai", output = "Cnidarian_stats.txt")
mixmodel(data = "Orbicella_faveolata_GCF_002042975_rna_eta_whole_seqs.txt", species = "Orbicella faveolata", output = "Cnidarian_stats.txt")
mixmodel(data="Physalia_physalis_eta_whole_seqs2.txt", species = "Physalia physalis", output = "Cnidarian_stats.txt")
mixmodel(data="Platygyra_carnosus_SRX118069_eta_whole_seqs2.txt", species = "Platygyra carnosus", output = "Cnidarian_stats.txt")
mixmodel(data = "Pocillopora_damicornis_GCF_003704095_rna_eta_whole_seqs.txt", species = "Pocillopora damicornis", output = "Cnidarian_stats.txt")
mixmodel(data="Podocoryna_carnea_eta_whole_seqs2.txt", species = "Podocoryna carnea", output = "Cnidarian_stats.txt")

mixmodel(data="Polypodum_hydriforme_SRR1336770_eta_whole_seqs2.txt", species = "Polypodum hydriforme", output = "Cnidarian_stats.txt")
mixmodel(data="Porpita_porpita_GHBA01_eta_whole_seqs2.txt", species = "Porpita_porpita", output = "Cnidarian_stats.txt")
mixmodel(data="Rhodactis_indosinensis_SRR3201278-81_eta_whole_seqs2.txt", species = "Rhodactis indosinensis", output = "Cnidarian_stats.txt")
mixmodel(data="Rhopilema_esculentum_GEMS01_eta_whole_seqs2.txt", species = "Rhopilema esculentum", output = "Cnidarian_stats.txt")
mixmodel(data="Ricordea_yuma_SRR3201251-3_eta_whole_seqs2.txt", species = "Ricordea yuma", output = "Cnidarian_stats.txt")
mixmodel(data="Seriatopora_hystrix_SRR2300678_eta_whole_seqs2.txt", species = "Seriatopora hystrix", output = "Cnidarian_stats.txt")
mixmodel(data="Stomolophus_meleagris_eta_whole_seqs2.txt", species = "Stomolophus meleagris", output = "Cnidarian_stats.txt")
mixmodel(data = "Stylophora_pistillata_GCF_002571385_rna_eta_whole_seqs.txt", species = "Stylophora pistillata (transcriptome)", output = "Cnidarian_stats.txt")
mixmodel(data = "Thelohanellus_kitauei_GCA_000827895_eta_whole_seqs.txt", species = "Thelohanellus kitauei (gene models)", output = "Cnidarian_stats.txt")
mixmodel(data="Tripedalia_cystophora_GHAQ01_eta_whole_seqs2.txt", species = "Tripedalia cystophora", output = "Cnidarian_stats.txt")
mixmodel(data="Turritopsis_sp_SK-2016_DRR053671-DRR053676_eta_whole_seqs2.txt", species = "Turritopsis sp.", output = "Cnidarian_stats.txt")
mixmodel(data="Velella_velella_GHAZ01_eta_whole_seqs2.txt", species = "Velella velella", output = "Cnidarian_stats.txt")
mixmodel(data="Xenia_sp_GHBC01_eta_whole_seqs2.txt", species = "Xenia sp.", output = "Cnidarian_stats.txt")


#old Morbakka, replaced by the new one above#
mixmodel(data="Morbakka_virulenta_GHAF01.1_rna_eta_whole_seqs.txt", species = "Morbakka virulenta", output = "Cnidarian_stats.txt")



### some stats for repeat CpG data ### 
data1 = read.table("/Users/zhangxinhui/Desktop/Morbakka_virulenta_GCA_003991215.1_MVIv1_repeatssJuly21_not_overlapping_genes_eta_whole_seqs.txt", header = TRUE, sep = " ", quote = "")
data1 = filter(data1, Length >= 50)
data1 = data1[data1$CGo_e!="Inf",]
mean(data1$CGo_e)
Repeat_CpG_stat = function(data,species,output){
  data1 = read.table(data, header = TRUE, sep = " ", quote = "")
  data1 = filter(data1, Length >= 50)
  data1 = data1[data1$TGo_e!="Inf" & data1$CGo_e!="Inf",]
  mean = mean(na.omit(data1$CGo_e))
  LinearModel = lm(data1$TGo_e ~ data1$CGo_e)
  pdf(file = paste(species, 'plot1'), width = 120, height = 50)
  par(mfrow=c(1,2))
  par(mai = c(6,8,6,1))
  par(cex = 7)
  plot(density(na.omit(data1$CGo_e)), main = species, xlab = "Normalized CpG content", ylab = "Density", lwd = 10)
  plot(x = data1$CGo_e, y = data1$TGo_e, xlim = c(0,2), ylim = c(0,3), xlab = "Normalized CpG content", ylab = "Normalized TpG content", pch = 20)
  abline(LinearModel, col = "red", lwd = 8)
  cat(paste(species, "\t", mean, "\t", coef(LinearModel)[1], "\t", coef(LinearModel)[2], "\n"), file=output, append = TRUE)
  dev.off()
}
Repeat_CpG_stat(data = "Acropora_digitefera_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Acropora digitefera (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Acropora_millepora_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Acropora millepora (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Actinia_tenebrosa_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Actinia tenebrosa (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Aiptasia_pallida_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Aiptasia pallida (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Dendronephthya_gigantea_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Dendronephthya gigantea (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Henneguya_salminicola_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Henneguya salminicola (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Hydra_vulgaris_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Hydra vulgaris (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Myxobolus_squamalis_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Myxobolus squamalis (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Orbicella_faveolata_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Orbicella faveolata (repeats)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "/Users/zhangxinhui/Desktop/Morbakka_virulenta_GCA_003991215.1_MVIv1_repeatssJuly21_not_overlapping_genes_eta_whole_seqs.txt", species = "Morbakka virulenta (repeats by me)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "/Users/zhangxinhui/Desktop/Aurelia.Genome_v1.2_11-27-18_repeatssJuly21_not_overlapping_genes_eta_whole_seqs.txt", species = "Aurelia coerulia (repeats by me)", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Thelohanellus_kitauei_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Thelohanellus kitauei", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Stylophora_pistillata_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Stylophora pistillata", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Pocillopora_damicornis_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Pocillopora damicornis", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")
Repeat_CpG_stat(data = "Nematostella_vectensis_repeats_not_overlapping_genes_eta_whole_seqs.txt", species = "Nematostella vectensis", output = "/Users/zhangxinhui/Box Sync/Aurelia_Genetics_Epigenetics/Methylation/CGcontent/Inverts_CpG/CpG_data/outputs/cnidaria/02_eta/Cnidarian_stats.txt")




